### PR TITLE
docs: Add Notational conventions topic

### PR DESCRIPTION
**Pull request by stoobie**
### Description of the Changes

This PR adds a page describing notational conventions in the docs.

### PR Preview URL

[Notational conventions](https://starknet-io.github.io/starknet-docs/pr-1242/documentation/notational-conventions)

### Check List

- [x] Changes have been done against main branch, and PR does not conflict
- [x] PR title follows the convention: `<docs/feat/fix/chore>(optional scope): <description>`, e.g: `fix: minor typos in code`

- - -
This change is [Reviewable](https://reviewable.io/reviews/starknet-io/starknet-docs/1242)

### CODE REVIEW

File: components/Starknet/modules/ROOT/pages/notational-conventions.adoc [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+| *bold* a| * Indicates GUI elements, such as button text or menu names.
+* Highlights text for added importance or to draw attention to it.
+
+| _italics_
+| Indicates the first occurrence of a new term, titles of documents or topics, and user-supplied or variable values.
+
+| `monospace` | Indicates code, commands, file paths, or other text that should be displayed in a fixed-width font to differentiate it from the surrounding text.
+
+| `<item>`
+| Indicates user-supplied or variable values in code, commands, file paths, or other text that should be displayed in a monospace font.
```
Comment (stoobie) [inline components/Starknet/modules/ROOT/pages/notational-conventions.adoc:17]:
```suggestion
| Indicates user-supplied or variable values in code, commands, file paths, or other text that should be displayed in a monospace font.

Can also refer to generic types. For example: `List<felt252>`
```

File: components/Starknet/modules/ROOT/pages/notational-conventions.adoc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+a| Square brackets indicate that the enclosed items are optional. Also can indicate a range of numbers, where the enclosed items are included. For example, in [0,100], 0 and 100 are part of the specified range.
+
+a| (item1, item2)
+| Parentheses indicate a range of numbers, where the enclosed items are not included. For example, in (0,100), 0 and 100 are _not_ part of the specified range.
+
+| item1 \| item2
+| A vertical bar (\|) indicates a choice between *item1* and *item2*.
+
+a| +{item1, item2}+
+| Curly brackets indicates a list or set of possible discrete values.
```
Comment (stoobie) [inline components/Starknet/modules/ROOT/pages/notational-conventions.adoc:35]:
```suggestion
| Curly brackets indicates a list or set of possible values.
```

File: components/Starknet/modules/ROOT/pages/notational-conventions.adoc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+| `monospace` | Indicates code, commands, file paths, or other text that should be displayed in a fixed-width font to differentiate it from the surrounding text.
+
+| `<item>`
+| Indicates user-supplied or variable values in code, commands, file paths, or other text that should be displayed in a monospace font.
+
+Can also refer to generic types. For example: `List<felt252>`
+
+| stem:[\text{string_in_math_notation}]
+| Indicates a fixed element or constant specified in mathematical notation.
```
Comment (stoobie) [inline components/Starknet/modules/ROOT/pages/notational-conventions.adoc:22]:
```suggestion
| A serif typeface indicates a fixed element or constant specified in mathematical notation.
```

File: components/Starknet/modules/ROOT/pages/notational-conventions.adoc [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+| `monospace` | Indicates code, commands, file paths, or other text that should be displayed in a fixed-width font to differentiate it from the surrounding text.
+
+| `<item>`
+| Indicates user-supplied or variable values in code, commands, file paths, or other text that should be displayed in a monospace font.
+
+Can also refer to generic types. For example: `List<felt252>`
+
+| stem:[\text{string_in_math_notation}]
+| A serif typeface indicates a fixed element or constant specified in mathematical notation.
```
Comment (stoobie) [inline components/Starknet/modules/ROOT/pages/notational-conventions.adoc:22]:
```suggestion
| A normal (non-italics) serif typeface indicates a fixed element or constant specified in mathematical notation.
```

**Comment by LandauRaz** [comment]
Sorry for the delay, LGTM.

**Review by LandauRaz — APPROVED** [review]

**Pull request merged**